controllers/etcd_controller_test.go

```diff
@@ -54,85 +90,229 @@ var _ = Describe("Druid", func() {
 			}
 			c.Create(context.TODO(), &ns)
 			storeSecret := instance.Spec.Backup.Store.SecretRef.Name
-			tlsClientSecret := instance.Spec.Etcd.TLS.ClientTLSSecretRef.Name
-			tlsServerName := instance.Spec.Etcd.TLS.ServerTLSSecretRef.Name
-			tlsCAName := instance.Spec.Etcd.TLS.TLSCASecretRef.Name
-			errors := createSecrets(c, instance.Namespace, storeSecret, tlsCAName, tlsClientSecret, tlsServerName)
+			errors := createSecrets(c, instance.Namespace, storeSecret)
 			Expect(len(errors)).Should(BeZero())
 		})
-		It("should create statefulset", func() {
+		It("should create and adopt statefulset", func() {
 			defer WithWd("..")()
-
 			err = c.Create(context.TODO(), instance)
 			Expect(err).NotTo(HaveOccurred())
-
 			ss := appsv1.StatefulSet{}
-			getStatefulset(c, instance, &ss)
-			testLog.Info("fetched Statefulset", "statefulset", ss.Name)
-			Eventually(ss.Name, timeout).ShouldNot(BeEmpty())
+			Eventually(func() error { return statefulsetIsCorrectlyReconcilced(c, instance, &ss) }, timeout, pollingInterval).Should(BeNil())
+			setStatefulSetReady(&ss)
+			err = c.Status().Update(context.TODO(), &ss)
+			Expect(err).NotTo(HaveOccurred())
 		})
 		AfterEach(func() {
 			c.Delete(context.TODO(), instance)
 		})
 	})
-	Context("when adding etcd resources with statefulset already present", func() {
-		var err error
-		var instance *druidv1alpha1.Etcd
-		var c client.Client
-		var ss *appsv1.StatefulSet
-		BeforeEach(func() {
-			instance = getEtcd("foo3", "default")
 
-			// Setup the Manager and Controller.  Wrap the Controller Reconcile function so it writes each request to a
-			// channel when it is finished.
+	DescribeTable("when adding etcd resources with statefulset already present",
+		func(name string, setupStatefulSet StatefulSetInitializer, setupPod bool) {
+			var ss *appsv1.StatefulSet
+			instance := getEtcd(name, "default", false)
+			c := mgr.GetClient()
+			switch setupStatefulSet {
+			case WithoutOwnerReference:
+				ss = createStatefulset(name, "default", instance.Spec.Labels)
+				Expect(ss.OwnerReferences).Should(BeNil())
+			case WithOwnerReference:
+				ss = createStatefulsetWithOwnerReference(instance)
+				Expect(len(ss.OwnerReferences)).ShouldNot(BeZero())
+			default:
```

Comment:
Should we have a case for pre-existing `StatefulSet` with annotations? I know this won't happen when we make the next release. But it would be good to cover the case which would happen for every reconciliation after that. WDYT?

Reply:
Pondering upon how the test would be like, what would be the expected behavior of the test. That the annotations still persist after the reconcile?